winch/codegen/src/isa/x64/masm.rs

```diff
@@ -30,15 +30,20 @@ use cranelift_codegen::{
         args::{ExtMode, CC},
         settings as x64_settings,
     },
-    settings, Final, MachBufferFinalized, MachLabel,
+    settings, Final, MachBufferFinalized, MachLabel, PatchRegion,
 };
 
+use smallvec::SmallVec;
 use wasmtime_environ::{PtrSize, WasmValType, WASM_PAGE_SIZE};
 
 /// x64 MacroAssembler.
 pub(crate) struct MacroAssembler {
     /// Stack pointer offset.
     sp_offset: u32,
+    /// Stack high-water mark.
+    sp_max: u32,
+    /// Regions where the addition of the used stack must be fixed up
```

Comment:
This could definitely be `1` or even an optional. Given that we could possibly call `check_stack` multiple times, I wanted to make sure that we would still generate valid code. Perhaps the right solution is an `Option<PatchRegion>` and an assert?

Reply:
Oh that makes sense, thanks for explaining. Yeah perhaps we could try going with `Option<PatchRegion>` + assert.